src/day8/re_frame/trace/app_state.cljs

```diff
@@ -11,12 +11,41 @@
        (reduce (fn [acc [property value]]
                  (assoc acc (keyword property) value)) {})))
 
+(defn data-input [{:keys [title on-save on-change on-stop]}]
+  (let [val  (r/atom title)
+        save #(let [v (-> @val str str/trim)]
+                (when (pos? (count v))
+                  (on-save v)))]
+    (fn []
+      [:input {:type        (cond (integer? "number")
+                                  (string?  "text"))
+               :value       @val
+               :size        (if (zero? (count (str @val)))
+                              1
+                              (count (str @val)))
+               :on-change   #(do (reset! val (-> % .-target .-value))
+                                 (on-change %))
+               :on-key-down #(case (.-which %)
+                               13 (save)
+                               nil)}])))
+
+(defn editable
+  [raw]
+  (let [val (r/atom raw)]
+    [:span {:class (str/join " " ["data-structure-editable"
+                                  (when (string? raw) "string")])}
+      [data-input {:title     raw
+                   :on-save   #(js/console.log %)
+                   :on-change #(reset! val (.. % -target -value))}]]))
+
+
 (defn str->hiccup
   [string]
-  (cond (= string "span")   :span
-        (= string "style")  :style
-        (= string ", ")     " "
-        :else               string))
+  (cond (= string "span")              :span
+        (= string "style")             :style
+        (= string ", ")                " "
+        (str/starts-with? string "\"") [editable (str/replace string "\"" "")]
```

Comment:
What's the purpose of the \? To mark it as editable? I don't love this approach, is there another way we can indicate that something is editable?

Reply:
cljs-devtools styles strings to be wrapped with `" "`. I want to strip the quotes when I put this in an input field so it doesn't get saved into app-state. I add them back in with css so that in the end, it visually matches the styles original intended by cljs-devtools. You can see this in the screenshots above (:

Reply:
Currently, the mark that something is editable is a dashed underline. Another possibility is that there could be no visible indicator, and then the dashed lines appear on click. I'm still playing around with styles!